monai/networks/nets/vltransformer.py

```diff
@@ -0,0 +1,342 @@
+        attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
+        attention_scores = attention_scores / math.sqrt(self.attention_head_size)
+        attention_probs = self.dropout(nn.Softmax(dim=-1)(attention_scores))
+        context_layer = torch.matmul(attention_probs, value_layer)
+        context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
+        context_layer = context_layer.view(*new_context_layer_shape)
+        return context_layer
+
+
+class BertOutput(nn.Module):
+    """BERT output layer.
+    Based on: BERT (pytorch-transformer)
+    https://github.com/huggingface/transformers
+    """
+
+    def __init__(self, config) -> None:
+        super(BertOutput, self).__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.LayerNorm = torch.nn.LayerNorm(config.hidden_size, eps=1e-12)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, hidden_states, input_tensor):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class BertMixedLayer(nn.Module):
+    """BERT cross attention layer.
+    Based on: BERT (pytorch-transformer)
+    https://github.com/huggingface/transformers
+    """
+
+    def __init__(
+        self,
+        config,
+    ) -> None:
+        super().__init__()
+        self.att = BertAttention(config)
+        self.output = BertOutput(config)
+
+    def forward(self, x, y):
+        output = self.att(x, y)
+        return self.output(output, x)
+
+
+class Pooler(nn.Module):
+    """BERT pooler layer.
+    Based on: BERT (pytorch-transformer)
+    https://github.com/huggingface/transformers
+    """
+
+    def __init__(
+        self,
+        hidden_size,
+    ) -> None:
+        super(Pooler, self).__init__()
+        self.dense = nn.Linear(hidden_size, hidden_size)
+        self.activation = nn.Tanh()
+
+    def forward(self, hidden_states):
+        first_token_tensor = hidden_states[:, 0]
+        pooled_output = self.dense(first_token_tensor)
+        pooled_output = self.activation(pooled_output)
+        return pooled_output
+
+
+class MultiModal(BertPreTrainedModel):
+    """
+    Multimodal Transformers From Pretrained BERT Weights"
+    """
+
+    def __init__(
+        self,
+        num_language_layers: int = 2,
+        num_vision_layers: int = 2,
+        num_mixed_layers: int = 2,
+    ) -> None:
+        """
+        Args:
+            num_language_layers: number of language transformer layers.
+            num_vision_layers: number of vision transformer layers.
+            num_mixed_layers: number of mixed transformer layers.
+        """
+        super().__init__()
+        bert_config = type(
```

Comment:
please make it a module-level variable `, like https://github.com/Project-MONAI/MONAI/blob/c50235c85dca411e878c0c3508eb27a0a76b37fb/monai/networks/nets/efficientnet.py#L28

Reply:
One issue here is that module-level variable does not work properly when trying to initialize model weights from bert checkpoint in huggingface library.

Reply:
how to replicate this issue, could you please elaborate?

Reply:
or you can make the dictionary (in line 232-255) a global variable as the default, 
```py
def __init__(
        self,
        num_language_layers: int = 2,
        num_vision_layers: int = 2,
        num_mixed_layers: int = 2,
        bert_config: Dict = BERT_CONFIG,
    )
```
it'll leave some flexibility for tuning the parameters for the end users.

Reply:
One way is to simply change the object-based definition to a dictionary-based config for BERT model, but `class BertPreTrainedModel ` is not compatible with this. I am using this class directly from Huggingface with very minor modifications. 

For your suggestion to give more flexibility, I think we need to keep the original BETR configurations to be able to use the pre-trained BERT weights (at least for this release). Pre-training a BERT type model on text data from scratch could be costly. Any suggestions ?

Reply:
are you sure that we need to keep exactly this version of config in order to load the pretrained?

Reply:
Basically, I first tried with different configurations. But the model will not be initialized from the BERT pre-trained checkpoint,so takes longer for training.

Reply:
my concern is mainly about the hard-coded config. for example, as a user I want to set ``attention_probs_dropout_prob`: 0.2` and `"transformers_version": "4.10.1" for fine-tuning of the pretrained model, how can I set this `MultiModal` class without rewriting everything from scratch?